modules/utxo-core/src/bip322/toSign.ts

```diff
@@ -65,14 +65,20 @@ export function buildToSignPsbtForChainAndIndex(
   if (isTaprootChain(chain)) {
     throw new Error('BIP322 is not supported for Taproot script types.');
   }
-  const output = bitgo.outputScripts.createOutputScript2of3(
-    rootWalletKeys.deriveForChainAndIndex(chain, index).publicKeys,
-    bitgo.scriptTypeForChain(chain)
-  );
+  const walletKeys = rootWalletKeys.deriveForChainAndIndex(chain, index);
+  const output = bitgo.outputScripts.createOutputScript2of3(walletKeys.publicKeys, bitgo.scriptTypeForChain(chain));
 
-  return buildToSignPsbt(message, {
+  const psbt = buildToSignPsbt(message, {
     scriptPubKey: output.scriptPubKey,
     redeemScript: output.redeemScript,
     witnessScript: output.witnessScript,
   });
+
+  psbt.updateInput(
+    0,
+    bitgo.getPsbtBip32DerivationOutputUpdate(rootWalletKeys, walletKeys, bitgo.scriptTypeForChain(chain))
+  );
+  bitgo.addXpubsToPsbt(psbt as bitgo.UtxoPsbt, rootWalletKeys);
```

Comment:
im noticing that buildToSignPsbt is doing this as bitgo.UtxoPsbt cast as well, https://github.com/BitGo/BitGoJS/blob/2641f250bb5ec7e8d0cbd12d146bc88d8a161df9/modules/utxo-core/src/bip322/toSign.ts#L26

is it possible to replace that line in toSign.ts with new UtxoPsbt instead of new Psbt? would that break anything?

Reply:
nothing really requires this to be a UtxoPsbt, which is more specific to 2of3, so I am keeping it this way. Maybe it is best to re-type those functions itself. Will do that